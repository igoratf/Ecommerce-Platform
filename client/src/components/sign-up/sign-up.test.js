import React from 'react';
import { shallow } from 'enzyme';

import { SignUp } from './sign-up.component';

describe('Sign Up component', () => {
  let wrapper;
  let mockSignUpStart;
  let mockIsSigningUp;

  beforeEach(() => {
    mockSignUpStart = jest.fn();
    mockIsSigningUp = false;

    const mockProps = {
      signUpStart: mockSignUpStart,
      isSigningUp: mockIsSigningUp
    }

    wrapper = shallow(<SignUp {...mockProps} />);
  });

  it('should render SignUp component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call mockSignUpStart when form is submitted', () => {
    wrapper.find('#signUpForm').simulate('submit', { preventDefault() {} });
    expect(mockSignUpStart).toHaveBeenCalled();
  });

  it('should render ButtonSpinner if isSigningUp is true', () => {
    const newMockProps = {
      signUpStart: mockSignUpStart,
      isSigningUp: true
    };

    const newWrapper = shallow(<SignUp {...newMockProps} />);

    expect(newWrapper.exists("ButtonSpinner")).toBe(true);
  })
})