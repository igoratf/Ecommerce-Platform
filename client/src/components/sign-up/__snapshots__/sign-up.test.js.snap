// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up component should render SignUp component 1`] = `ShallowWrapper {}`;
